rest/server_context.go

```diff
@@ -1147,6 +1149,68 @@ func (sc *ServerContext) updateCalculatedStats() {
 
 }
 
+func initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPath string) (*gocbcore.Agent, error) {
+	authenticator, err := base.GoCBCoreAuthConfig(clusterUser, clusterPass, certPath, keyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config := gocbcore.AgentConfig{
+		Auth:           authenticator,
+		AuthMechanisms: []gocbcore.AuthMechanism{gocbcore.ScramSha512AuthMechanism},
+		TLSRootCAProvider: func() *x509.CertPool {
+			return nil
+		},
+	}
+
+	err = config.FromConnStr(clusterAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	agent, err := gocbcore.CreateAgent(&config)
+	if err != nil {
+		return nil, err
+	}
+
+	agentReadyErr := make(chan error)
+	_, err = agent.WaitUntilReady(time.Now().Add(5*time.Second), gocbcore.WaitUntilReadyOptions{}, func(result *gocbcore.WaitUntilReadyResult, err error) {
```

Comment:
5 seconds seems low enough to be flaky under some conditions (test env?).
Is this a deadline for the internal retry loop, rather than a fixed sleep? If so, I think we can increase this a bit (15 seconds?)

Reply:
We only care about waiting for the `gocbcore.MgmtService`, otherwise it's apparently going to try and wait for everything: `ServiceTypes []ServiceType // Defaults to all services`

Reply:
Yeah its a deadline. Was just concerned that a HTTP call could just block for 15 seconds but I can go with that.
Will set Mgmt service. I guess it knows whats enabled and whats not then? Given my cluster had only KV and Query but didn't block?